_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -173,6 +173,132 @@ You can find detailed explanations, runnable examples and tips to maintain binar
 
 Again, we recommend using MiMa to double-check that you have not broken binary compatibility after making changes.
 
+### Changing a case class definition in a backwards-compatible manner
+
+Sometimes, it is desirable to change the definition of a case class (adding and/or removing fields) while still staying backwards-compatible with the existing usage of the case class, i.e. not breaking the so-called _binary compatibility_. The first question you should ask yourself is “do you need a _case_ class?” (as opposed to a regular class, which can be easier to evolve in a binary compatible way). A good reason for using a case class is when you need a structural implementation of `equals` and `hashCode`.
+
+To achieve that, follow this pattern:
+ * make the primary constructor private (this makes private the `copy` method of the class)
+ * define a private `unapply` function in the companion object (note that by doing that the case class loses the ability to be used as an extractor in match expressions)
+ * for all the fields, define `withXXX` methods on the case class that create a new instance with the respective field changed (you can use the private `copy` method to implement them)
+ * create a public constructor by defining an `apply` method in the companion object (it can use the private constructor)
+
+Example:
+
+{% tabs case_class_compat_1 class=tabs-scala-version %}
+
+{% tab 'Scala 2' %}
+~~~ scala
+// Mark the primary constructor as private
+case class Person private (name: String, age: Int) {
+  // Create withXxx methods for every field, implemented by using the copy method
```

Comment:
Scala 2 is a whole different game. I think we also have to make the `copy` method private. And consistently across the changes to the case class.

Reply:
Oh, good point, I forgot that `copy` is not private in Scala 2. I’ve reverted the changes that included Scala 2 then.